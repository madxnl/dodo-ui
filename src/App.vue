
<template>
  <HelloWorld msg="Hello Vue 3 + TypeScript + Vite" />
</template>

<script setup lang="ts">
import HelloWorld from './components/HelloWorld.vue';
</script>

<style>
html, body {
  margin: 0;
  height: 100%;
}
#app {
  background: skyblue;
  height: 100%;
  display: flex;
  flex-flow: column;
  overflow: auto;
}
</style>
