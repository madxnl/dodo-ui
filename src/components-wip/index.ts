export { default as NavLayout } from './NavLayout.vue'
export { default as Spinner } from './Spinner.vue'
