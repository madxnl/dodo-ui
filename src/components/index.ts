export { default as Animated } from './Animated.vue'
export { default as Avatar } from './Avatar.vue'
export { default as Button } from './Button.vue'
export { default as Card } from './Card.vue'
export { default as Checkbox } from './Checkbox.vue'
export { default as Chip } from './Chip.vue'
export { default as ChoiceChips } from './ChoiceChips.vue'
export { default as Column } from './Column.vue'
export { default as CrashDialog } from './CrashDialog.vue'
export { default as Datatable } from './Datatable.vue'
export { default as Dialog } from './Dialog.vue'
export { default as Dropdown } from './Dropdown.vue'
export { default as FieldOld } from './FieldOld.vue'
export { default as FluidGrid } from './FluidGrid.vue'
export { default as Form } from './Form.vue'
export { default as FormItem } from './FormItem.vue'
export { default as Grid } from './Grid.vue'
export { default as Icon } from './Icon.vue'
export { default as MenuItem } from './MenuItem.vue'
export { default as Modal } from './Modal.vue'
export { default as NavBar } from './NavBar.vue'
export { default as NavBarItem } from './NavBarItem.vue'
export { default as NavBarNew } from './NavBarNew.vue'
export { default as NavItem } from './NavItem.vue'
export { default as Row } from './Row.vue'
export { default as ScrollContainer } from './ScrollContainer.vue'
export { default as SelectOld } from './SelectOld.vue'
export { default as Select } from './SimpleSelect.vue'
export { default as SocialIcon } from './SocialIcon.vue'
export { default as Spinner } from './Spinner.vue'
export { default as Stepper } from './Stepper.vue'
export { default as Tabs } from './Tabs.vue'
export { default as TextInput } from './TextInput.vue'
export { default as Textarea } from './Textarea.vue'
export { default as Tooltip } from './Tooltip.vue'
