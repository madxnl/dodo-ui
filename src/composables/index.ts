export * from './composables'
export * from './useServiceFactory'
export * from './useStyle'
