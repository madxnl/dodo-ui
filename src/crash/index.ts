export { default as CrashDialog } from './CrashDialog.vue'
export { crashDialogPlugin } from './crashInfo'
