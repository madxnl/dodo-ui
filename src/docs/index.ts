export { default as DocsComponentSection } from './DocsComponentSection.vue'
export { default as DocsExample } from './DocsExample.vue'
