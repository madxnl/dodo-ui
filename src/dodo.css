@import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;500&display=swap');

:root {
  --dodo-spacing: 4px;

  --dodo-rgb-info: 58, 134, 255;
  --dodo-rgb-success: 0, 177, 136;
  --dodo-rgb-warn: 223, 178, 53;
  --dodo-rgb-warning: var(--dodo-rgb-warn);
  --dodo-rgb-danger: 255, 0, 83;
  --dodo-rgb-background: 255, 255, 255;
  --dodo-rgb-foreground: 9, 30, 40;
  --dodo-rgb-primary: var(--dodo-rgb-info);
  --dodo-rgb-secondary: var(--dodo-rgb-success);

  --dodo-color-info: rgb(var(--dodo-rgb-info));
  --dodo-color-success: rgb(var(--dodo-rgb-success));
  --dodo-color-warn: rgb(var(--dodo-rgb-warn));
  --dodo-color-danger: rgb(var(--dodo-rgb-danger));
  --dodo-color-background: rgb(var(--dodo-rgb-background));
  --dodo-color-foreground: rgb(var(--dodo-rgb-foreground));
  --dodo-color-primary: rgb(var(--dodo-rgb-primary));
  --dodo-color-secondary: rgb(var(--dodo-rgb-secondary));

  --dodo-font-size: 14px;
  --dodo-font-small: 13px;
  --dodo-weight-body: 400;
  --dodo-weight-bold: 500;
  --dodo-font-base: var(--dodo-font-size) Inter, sans-serif;

  --dodo-button-height: 36px;
  --dodo-transition-duration: 0.15s;

  --dodo-breakpoint-small: 600;
  --dodo-breakpoint-large: 1200;
  --dodo-gap-0: 0px;
  --dodo-gap-1: calc(var(--dodo-spacing) * 1);
  --dodo-gap-2: calc(var(--dodo-spacing) * 2);
  --dodo-gap-3: calc(var(--dodo-spacing) * 3);
  --dodo-gap-4: calc(var(--dodo-spacing) * 4);
  --dodo-gap-5: calc(var(--dodo-spacing) * 5);
  --dodo-gap-6: calc(var(--dodo-spacing) * 6);
  --dodo-gap-8: calc(var(--dodo-spacing) * 8);
  --dodo-gap-10: calc(var(--dodo-spacing) * 10);
  --dodo-gap-12: calc(var(--dodo-spacing) * 12);
  --dodo-gap-16: calc(var(--dodo-spacing) * 16);
  --dodo-gap-24: calc(var(--dodo-spacing) * 24);
  --dodo-gap-32: calc(var(--dodo-spacing) * 32);

  --dodo-shadow-1: 0px 1px 1px rgba(0, 0, 0, 0.1);
  --dodo-shadow-2: 0px 3px 3px rgba(0, 0, 0, 0.1);
  --dodo-shadow-3: 0px 6px 6px rgba(0, 0, 0, 0.1);
  --dodo-shadow-4: 0px 16px 16px rgba(0, 0, 0, 0.1);
  --dodo-shadow-5: 0px 32px 40px rgba(0, 0, 0, 0.1);
}

/* Layout */
.dodo-gap-0 {
  gap: 0 !important;
}
.dodo-gap-1 {
  gap: var(--dodo-spacing) !important;
}
.dodo-gap-2 {
  gap: var(--dodo-gap-2) !important;
}
.dodo-gap-4 {
  gap: var(--dodo-gap-4) !important;
}
.dodo-gap-5 {
  gap: var(--dodo-gap-5) !important;
}
.dodo-gap-6 {
  gap: var(--dodo-gap-6) !important;
}
.dodo-gap-8 {
  gap: var(--dodo-gap-8) !important;
}
.dodo-gap-12 {
  gap: var(--dodo-gap-12) !important;
}
.dodo-gap-16 {
  gap: var(--dodo-gap-16) !important;
}
.dodo-gap-24 {
  gap: var(--dodo-gap-24) !important;
}
.dodo-gap-32 {
  gap: var(--dodo-gap-32) !important;
}

.dodo-pad-0 {
  padding: 0 !important;
}
.dodo-pad-1 {
  padding: var(--dodo-spacing) !important;
}
.dodo-pad-2 {
  padding: var(--dodo-gap-2) !important;
}
.dodo-pad-4 {
  padding: var(--dodo-gap-4) !important;
}
.dodo-pad-5 {
  padding: var(--dodo-gap-5) !important;
}
.dodo-pad-6 {
  padding: var(--dodo-gap-6) !important;
}
.dodo-pad-8 {
  padding: var(--dodo-gap-8) !important;
}
.dodo-pad-12 {
  padding: var(--dodo-gap-12) !important;
}
.dodo-pad-16 {
  padding: var(--dodo-gap-16) !important;
}
.dodo-pad-24 {
  padding: var(--dodo-gap-24) !important;
}
.dodo-pad-32 {
  padding: var(--dodo-gap-32) !important;
}

.dodo-justify-start {
  justify-content: start !important;
}
.dodo-justify-end {
  justify-content: end !important;
}
.dodo-justify-center {
  justify-content: safe center !important;
}
.dodo-justify-between {
  justify-content: space-between !important;
}
.dodo-justify-around {
  justify-content: space-around !important;
}
.dodo-justify-evenly {
  justify-content: space-evenly !important;
}

.dodo-align-start {
  align-items: start !important;
}
.dodo-align-end {
  align-items: end !important;
}
.dodo-align-center {
  align-items: safe center !important;
}
.dodo-align-baseline {
  align-items: baseline !important;
}
.dodo-align-stretch {
  align-items: stretch !important;
}

.dodo-grow {
  flex-grow: 1 !important;
}

.dodo-wrap {
  flex-wrap: wrap !important;
}

/* Text */

.dodo-nowrap {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}
.dodo-fade-secondary,
.dodo-opacity-secondary {
  opacity: 0.6;
}
.dodo-fade-disabled,
.dodo-opacity-disabled {
  opacity: 0.38;
}
.dodo-text-center,
.dodo-textalign-center {
  text-align: center;
}
.dodo-text-end,
.dodo-textalign-end {
  text-align: end;
}
.dodo-color-info {
  color: var(--dodo-color-info);
}
.dodo-color-success {
  color: var(--dodo-color-success);
}
.dodo-color-warn {
  color: var(--dodo-color-warn);
}
.dodo-color-danger {
  color: var(--dodo-color-danger);
}
.dodo-color-background {
  color: var(--dodo-color-background);
}
.dodo-color-foreground {
  color: var(--dodo-color-foreground);
}
.dodo-color-primary {
  color: var(--dodo-color-primary);
}
.dodo-color-secondary {
  color: var(--dodo-color-secondary);
}

/* Font */

.dodo-fonts {
  font: var(--dodo-font-base);
  color: var(--dodo-color-foreground);
  overflow-wrap: break-word;
  hyphens: auto;
}
:where(.dodo-fonts) p {
  margin: 0;
  line-height: 1.4;
}
:where(.dodo-fonts) h1,
:where(.dodo-fonts) h2,
:where(.dodo-fonts) h3,
:where(.dodo-fonts) h4,
:where(.dodo-fonts) h5,
:where(.dodo-fonts) h6 {
  font-weight: var(--dodo-weight-bold);
  color: var(--dodo-rgb-foreground);
  font-size: var(--dodo-font-size);
  margin: 0;
}
:where(.dodo-fonts) h1 {
  font-size: calc(var(--dodo-font-size) * 2.5);
}
:where(.dodo-fonts) h2 {
  font-size: calc(var(--dodo-font-size) * 2);
}
:where(.dodo-fonts) h3 {
  font-size: calc(var(--dodo-font-size) * 1.5);
}
:where(.dodo-fonts) h4 {
  font-size: calc(var(--dodo-font-size) * 1.25);
}
:where(.dodo-fonts) h5 {
  font-size: calc(var(--dodo-font-size) * 1.15);
}
:where(.dodo-fonts) code {
  font-family: monospace;
  font-size: calc(var(--dodo-font-size) * 1.05);
}
:where(.dodo-fonts) a {
  color: var(--dodo-color-info);
  cursor: pointer;
  text-decoration: none;
}
:where(.dodo-fonts) blockquote {
  margin: 0;
  padding-left: 16px;
  border-left: 2px solid rgba(var(--dodo-rgb-foreground), 0.25);
}
:where(.dodo-fonts) small {
  font-size: var(--dodo-font-small) !important;
}
:where(.dodo-fonts) strong {
  font-weight: var(--dodo-weight-bold) !important;
}
:where(.dodo-fonts) label {
  margin: 0;
}
:where(.dodo-fonts) ::selection {
  color: white;
  background: var(--dodo-color-info);
}
:where(.dodo-fonts) th {
  font-weight: var(--dodo-weight-bold);
  font-size: var(--dodo-font-small);
  text-align: start;
}
:where(.dodo-fonts) table {
  border-collapse: collapse;
}
:where(.dodo-fonts) hr {
  margin: 0;
  border: 0;
  min-width: 1px;
  min-height: 1px;
  background: currentColor;
  opacity: 0.1;
}
:where(.dodo-fonts) mark {
  background: rgba(var(--dodo-rgb-info), 0.2);
  padding: 0 2px;
  border-radius: 4px;
}

/* Fields */

.dodo-formfield {
  min-width: 0;
  padding: 2px 8px;
  box-shadow: 0 1px 2px inset rgba(0, 0, 0, 0.1);
  background: white;
  border: 1px solid rgba(var(--dodo-rgb-foreground), 0.3);
  border-radius: 4px;
  --height: var(--dodo-button-height);
  min-height: var(--height);
  font: var(--dodo-font-base);
  color: var(--dodo-color-foreground);
  box-sizing: border-box;
}
.dodo-formfield:focus {
  border-color: var(--dodo-color-info);
  outline: 1px solid var(--dodo-color-info);
}
.dodo-formfield:disabled {
  opacity: 0.38;
  cursor: not-allowed;
}
