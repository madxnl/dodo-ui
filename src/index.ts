// This is the index for the NPM package
export { default as Avatar } from './avatar/Avatar.vue'
export { default as Button } from './button/Button.vue'
export { default as Chip } from './button/Chip.vue'
export { default as Card } from './card/Card.vue'
export { default as Datatable } from './datatable/DatatableTyped.vue'
export { default as CrashDialog } from './dialog/CrashDialog.vue'
export { default as Dialog } from './dialog/Dialog.vue'
export { default as Dropdown } from './dropdown/Dropdown.vue'
export { default as DropdownItem } from './dropdown/DropdownItem.vue'
export { default as Field } from './form/Field.vue'
export { default as Form } from './form/Form.vue'
export { default as Label } from './form/Label.vue'
export { default as Icon } from './icon/Icon.vue'
export * from './icon/iconNames'
export { default as SocialIcon } from './icon/SocialIcon.vue'
export { default as Checkbox } from './input/Checkbox.vue'
export { default as TextInput } from './input/TextInput.vue'
export { default as Column } from './layout/Column.vue'
export { default as FluidGrid } from './layout/FluidGrid.vue'
export { default as Row } from './layout/Row.vue'
export { default as ScrollContainer } from './layout/ScrollContainer.vue'
export { default as NavBar } from './navbar/NavBar.vue'
export { default as NavBarItem } from './navbar/NavBarItem.vue'
export { default as Select } from './select/Select.vue'
export { default as SelectButtons } from './select/SelectButtons.vue'
export { default as Spinner } from './spinner/Spinner.vue'
export { default as Tabs } from './tabs/Tabs.vue'
export { default as Text } from './text/Text.vue'
export * from './theme'
export { default as Tooltip } from './tooltip/Tooltip.vue'
