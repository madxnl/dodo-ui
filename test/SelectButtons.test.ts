import { mount } from '@vue/test-utils'
import { SelectButtons } from '../src'
import { test, expect } from 'vitest'

const _options = [
  { label: 'Option one', value: 1 },
  { label: 'Option two', value: 2 },
  { label: 'Option three', value: 3 },
]

/**
* @vitest-environment jsdom
*/

test('Select a value', async () => {
  const wrapper = mount(SelectButtons, {
    propsData: { options: _options, modelValue: 1 },
  })
  await wrapper.findAll('button').find(w => w.text().includes('Option two'))!.trigger('click')
  expect(wrapper.emitted()).toMatchObject({ 'update:modelValue': [[2]] })
  expect(wrapper.html()).toMatchSnapshot()
})

test('Select multiple', async () => {
  const wrapper = mount(SelectButtons, {
    propsData: { options: _options, modelValue: [1], multiple: true },
  })
  await wrapper.findAll('button').find(w => w.text().includes('Option two'))!.trigger('click')
  expect(wrapper.emitted()).toMatchObject({ 'update:modelValue': [[[1, 2]]] })
  expect(wrapper.html()).toMatchSnapshot()
})

test('Deselect', async () => {
  const wrapper = mount(SelectButtons, {
    propsData: { options: _options, modelValue: [1, 2], multiple: true },
  })
  await wrapper.get('button').trigger('click')
  expect(wrapper.emitted()).toMatchObject({ 'update:modelValue': [[[2]]] })
  expect(wrapper.html()).toMatchSnapshot()
})
