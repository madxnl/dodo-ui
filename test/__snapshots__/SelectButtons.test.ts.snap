// Vitest Snapshot v1

exports[`Deselect 1`] = `
"<div wrap=\\"true\\" class=\\"dodoSelectButtons-Select dodoSelectButtons-multiple\\"><button style=\\"--bnt-rgb: 58,134,255;\\" type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-active dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option one
    </div>
    <!--v-if-->
  </button><button style=\\"--bnt-rgb: 58,134,255;\\" type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-active dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option two
    </div>
    <!--v-if-->
  </button><button type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option three
    </div>
    <!--v-if-->
  </button></div>"
`;

exports[`Select a value 1`] = `
"<div wrap=\\"false\\" class=\\"dodoSelectButtons-Select\\"><button style=\\"--bnt-rgb: 58,134,255;\\" type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-active dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option one
    </div>
    <!--v-if-->
  </button><button type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option two
    </div>
    <!--v-if-->
  </button><button type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option three
    </div>
    <!--v-if-->
  </button></div>"
`;

exports[`Select multiple 1`] = `
"<div wrap=\\"true\\" class=\\"dodoSelectButtons-Select dodoSelectButtons-multiple\\"><button style=\\"--bnt-rgb: 58,134,255;\\" type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-active dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option one
    </div>
    <!--v-if-->
  </button><button type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option two
    </div>
    <!--v-if-->
  </button><button type=\\"button\\" class=\\"dodoButton dodoButton-default dodoButton-rounded\\">
    <div class=\\"dodoButton-content\\">
      <!--
        @slot Menu Item footer
        @binding {object} option icon of the menu item
        @binding {string} index text of the menu item
      -->Option three
    </div>
    <!--v-if-->
  </button></div>"
`;
